packages/react/src/auto/hooks/useModelActionsForTable.tsx

```diff
@@ -0,0 +1,67 @@
+import React, { useMemo } from "react";
+import deepEqual from "react-fast-compare";
+import { humanizeCamelCase } from "../../utils.js";
+
+export type ModelActionDetails = {
+  apiIdentifier: string;
+  operationName: string;
+  isDeleter: boolean;
+  isBulk: boolean;
+  acceptsModelInput: boolean;
+  variables: any;
+};
+
+/**
+ * Returns a list of ModelActionDetail objects for given model
+ */
+const useModelActionsForTable = (props: { model: any }) => {
+  const modelActions: ModelActionDetails[] = [];
+
+  const modelPrototype = Object.getPrototypeOf(props.model);
+  for (const key of Object.keys(modelPrototype)) {
+    const isAction = modelPrototype[key].type === "action";
+    if (isAction) {
+      modelActions.push(modelPrototype[key]);
+    }
+  }
+
+  const funcMapper = (func: any): ModelActionDetails => ({
+    apiIdentifier: func.functionName,
+    operationName: func.operationName,
+    isDeleter: func.isDeleter,
+    isBulk: func.isBulk,
+    acceptsModelInput: func.acceptsModelInput,
+    variables: func.variables,
+  });
+
+  return modelActions.map(funcMapper);
+};
+
+export const useTableBulkActions = (props: { model: any }) => {
```

Comment:
I'm not sure I fully understand the desired usage here -- Why does this keep the selected model action details in state?

Reply:
This hook was made to provide bulk action options and track which one has been selected so that the modal can be opened with the corresponding action info.